AcademyApp/ViewControllers/LoginViewController.swift

```diff
@@ -0,0 +1,189 @@
+//
+//  ViewController.swift
+//  AcademyApp
+//
+//  Created by Peter Sevcik on 13/06/2019.
+//  Copyright © 2019 Peter Sevcik. All rights reserved.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController
+{
+    // MARK: - View Controller's life cycle methods
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        addObservers()
+        setupUI()
+        emailTextField.delegate = self
+        passwordTextField.delegate = self
+        
+        defaultDoneButtonBottomSpacing = doneButtonBottomConstraint.constant
+    }
+
+    deinit {
+        removeObservers()
+    }
+    
+    // MARK: - Stored Properities and methods
+    var defaultDoneButtonBottomSpacing : CGFloat!
+    
+    func isValidEmail(possibleEmail text: String)->Bool {
+        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]+"
+        let emailTest = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
+        return emailTest.evaluate(with: text)
+    }
+    
+    // MARK: - @IBOutlets @IBActions
+    @IBOutlet weak var doneButtonBottomConstraint: NSLayoutConstraint!
+    @IBOutlet var horizontalLineViews: [UIView]!
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    @IBOutlet weak var doneButton: UIButtonX!
+    
+    @IBOutlet weak var showButton: UIButton!
+    
+    @IBAction func backgroundTapped(_ sender: UITapGestureRecognizer) {
+        self.view.endEditing(true)
+    }
+    
+    @IBAction func emailTextFieldEditingDidChange(_ sender: UITextField) {
+        doneButton.isEnabled = isValidEmail(possibleEmail: sender.text!) && passwordTextField.text!.count > 0 ? true : false
+    }
+    
+    @IBAction func passwordTextFieldEditingDidChange(_ sender: UITextField) {
+        UIView.animate(withDuration: 0.2) {
+            self.showButton.isHidden = sender.text!.count > 0 ? false : true
+        }
+        
+        doneButton.isEnabled = isValidEmail(possibleEmail: emailTextField.text!) && passwordTextField.text!.count > 0 ? true : false
+    }
+    
+    // control event : touch down (hold)
+    @IBAction func showPassword(_ sender: UIButton) {
+        passwordTextField.isSecureTextEntry = false
+    }
+    
+    // control event : touch up inside (release)
+    @IBAction func hidePassword(_ sender: UIButton) {
```

Comment:
@IBActions should be named according to what triggers them - didTapOnButton, didTouchUpInsideButton, etc. - not hidePassword/showPassword

Reply:
Joined show/hide functionality into one method - showButtonTapped